test/solvers.jl

```diff
@@ -12,19 +12,28 @@
 # Should be run as part of runtests.jl
 #############################################################################
 
+function package_installed(name::Symbol)
+    try
+        @eval import $name
+        return true
+    catch e
+        return false
+    end
+end
+
 # Detect solvers
-grb = isdir(Pkg.dir("Gurobi"))
-cpx = isdir(Pkg.dir("CPLEX"))
-mos = isdir(Pkg.dir("Mosek"))
-cbc = isdir(Pkg.dir("Cbc"))
-glp = isdir(Pkg.dir("GLPKMathProgInterface"))
-ipt = isdir(Pkg.dir("Ipopt"))
-nlo = isdir(Pkg.dir("NLopt"))
-kni = isdir(Pkg.dir("KNITRO"))
-eco = isdir(Pkg.dir("ECOS"))
-osl = isdir(Pkg.dir("CoinOptServices"))
-scs = isdir(Pkg.dir("SCS"))
-nlw = isdir(Pkg.dir("AmplNLWriter"))
+grb = package_installed(:Gurobi)
+cpx = package_installed(:CPLEX)
+mos = package_installed(:Mosek)
+cbc = package_installed(:Cbc)
+glp = package_installed(:GLPKMathProgInterface)
+ipt = package_installed(:Ipopt)
+nlo = package_installed(:NLopt)
+kni = package_installed(:KNITRO)
+eco = package_installed(:ECOS)
+osl = package_installed(:CoinOptServices)
+scs = package_installed(:SCS)
+nlw = package_installed(:AmplNLWriter)
 # Load them
 if grb; import Gurobi; end
```

Comment:
I renamed the function to make it clear that it actually load the package instead of just checking its existence and removed most of the import below.